src/smc_pyutil/smc_pyutil/smc_open.py

```diff
@@ -50,7 +50,8 @@ def main():
         print "Opens each file (or directory) in the Sagemath Cloud web-based editor from the shell."
         print "If the named file doesn't exist, it is created."
     else:
-        process(sys.argv[1:])
+        # ignore args that have ? or * after glob expansion
+        process([p for p in sys.argv[1:] if re.search('[\?\*]', p) is None])
```

Comment:
Yep, "what if my filename really has a question mark or star?" -- that was my "IMPORTANT" point 3 below too.